flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNHighAvailabilityITCase.java

```diff
@@ -124,24 +135,84 @@ public void testKillYarnSessionClusterEntrypoint() throws Exception {
 
 		final ApplicationId id = restClusterClient.getClusterId();
 
-		waitUntilJobIsRunning(restClusterClient, jobId, RETRY_TIMEOUT);
+		waitUntilJobIsRunning(restClusterClient, jobId);
 
 		killApplicationMaster(yarnClusterDescriptor.getYarnSessionClusterEntrypoint());
+		waitForApplicationAttempt(id, 2);
 
+		waitUntilJobIsRunning(restClusterClient, jobId);
+
+		killApplicationAndWait(id);
+	}
+
+	@Test
+	public void testJobRecoversAfterKillingTaskManager() throws Exception {
+		final RestClusterClient<ApplicationId> restClusterClient = deploySessionCluster(setupYarnClusterDescriptor());
+		final JobID jobId = submitJob(restClusterClient, createJobGraph());
+		final ApplicationId id = restClusterClient.getClusterId();
+		waitUntilJobIsRunning(restClusterClient, jobId);
+
+		stopTaskManagerContainer();
+		waitUntilJobIsRunning(restClusterClient, jobId);
```

Comment:
Can it happen that the JobManager does not realize that a `TM` has died before we call `waitUntilJobIsRunning`? Is there an easy way to check whether we have restarted the job?